### PR TITLE
avoid using regex when building PATH (closes #2079)

### LINKED ISSUE #2079: Not able to start RStudio, The error occurred while parsing the regular expression fragment
### System details

    RStudio Edition : 
    RStudio Version : RStudio-1.1.419
    OS Version     high sierra 10.13.2
    R Version     R version 3.4.3

### Steps to reproduce the problem & Describe the problem in detail

when I installed and clicked the Rstudio icon, Rstudio gives a blank window and a dialog box saying below:

"Unexpected exception: Invalid preceding regular expression prior to repetition operator.  The error occurred while parsing the regular expression fragment: '(^|:)**>>>HERE>>> python mo'."   

it just does not work at all.  I firstly try to search from internet with no helps. 

Thank you for taking the time to file an issue!  -->

### PR DESCRIPTION
**Pull request by kevinushey**
Previously, we were constructing regular expressions based on entries within files located within `/etc/paths.d`, and this could cause failures to launch if files with entries that were not valid regular expressions were present.

The fix here is to avoid the use of regular expressions and just handle the path entries as regular strings.

### CODE REVIEW

**Comment by kevinushey** [comment]
Rebasing to v1.3 didn't go so well ... let me try again.

**Comment by kevinushey** [comment]
Should be clean for merging to v1.3 now.

File: src/cpp/session/modules/SessionPath.cpp [hunk 1]
```diff
@@ -66,19 +66,19 @@ void safeReadPathsFromFile(const FilePath& filePath,
...
-         if (!pPath->empty())
-            pPath->push_back(':');
-         pPath->append(entry);
-      }
-   }
-   CATCH_UNEXPECTED_EXCEPTION;
+   auto it = std::find(
+            pPathComponents->begin(),
+            pPathComponents->end(),
+            entry);
```
Comment (jmcphers) [inline src/cpp/session/modules/SessionPath.cpp]: It looks like we used to tolerate a trailing slash in the entry (i.e. `foo` matched `foo/`) but don't any more -- intentional?

Comment (kevinushey) [reply src/cpp/session/modules/SessionPath.cpp]: That wasn't intentional -- I'll fix that up.
